addons/api/tests/unit/utils/flatten-nested-object-test.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { flattenObject } from 'api/utils/flatten-nested-object';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | flatten-nested-object', function () {
+  test('it flattens a nested object to single level object', function (assert) {
+    const input = {
+      a: 1,
+      b: { c: 2, d: 3 },
+    };
+    const expected = {
+      a: 1,
+      'b.c': 2,
+      'b.d': 3,
+    };
+    assert.deepEqual(flattenObject(input), expected);
+  });
+
+  test('removes keys with null/empty values', function (assert) {
+    const input = {
+      a: null,
+      b: { c: 2 },
```

Comment:
Can we add a an object that is null at first level?
```
const input = {
      b: { c: null },
    };
```
Just to cover the case when receiving a null object.